### PR TITLE
HHH-15090 Access to public field with extended bytecode enhancement returns null for entity lazy-loaded from polymorphic toOne association

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net/browse/HHH-15090

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/proxy/pojo/bytebuddy/ByteBuddyProxyHelper.java [hunk 1]
```diff
@@ -52,29 +58,46 @@ public Class buildProxy(
...
-				.method( byteBuddyState.getProxyDefinitionHelpers().getVirtualNotFinalizerFilter() )
-						.intercept( byteBuddyState.getProxyDefinitionHelpers().getDelegateToInterceptorDispatcherMethodDelegation() )
-				.method( byteBuddyState.getProxyDefinitionHelpers().getHibernateGeneratedMethodFilter() )
+				.method( helpers.getVirtualNotFinalizerFilter() )
+						.intercept( helpers.getDelegateToInterceptorDispatcherMethodDelegation() )
+				.method( helpers.getHibernateGeneratedMethodFilter()
+						// HHH-15090: Don't apply extended enhancement reader/writer methods to the proxy;
+						// those need to be executed on the actual entity.
+						.and( ElementMatchers.not( helpers.getHibernateGeneratedFieldReaderMethodFilter() ) )
+						.and( ElementMatchers.not( helpers.getHibernateGeneratedFieldWriterMethodFilter() ) ) )
```
[the remark this answers is not in the document]
Comment (gsmet) [reply hibernate-core/src/main/java/org/hibernate/proxy/pojo/bytebuddy/ByteBuddyProxyHelper.java:96]: That's why we have this ugly `helpers` pattern btw.

[the remark this answers is not in the document]
Comment (yrodiere) [reply hibernate-core/src/main/java/org/hibernate/proxy/pojo/bytebuddy/ByteBuddyProxyHelper.java]: Done!

**Review by beikov — APPROVED** [review]
LGTM

**Comment by yrodiere** [comment]
Merged, thanks

**Pull request merged**